books/RayTracingTheNextWeek.html

```diff
@@ -2256,7 +2256,319 @@
+geometric primitive. Time to add our second primitive: the quadrilateral.
+
+
+Ray-Plane Intersection
+-----------------------
+Spheres are generally the first ray tracing primitive because their implicit formula makes it so
+easy to solve for ray intersection. Like spheres, planes also have an implicit formula, and
+ray-plane intersection is even _easier_ to solve.
+
+You may already know this implicit formula for a plane:
+
+  $$ Ax + By + Cz + D = 0 $$
+
+That is, a plane is the set of all points $\mathbf{P} = (x,y,z)$ that satisfy the formula above. It
+makes things slightly easier to use the alternate formulation:
+
+  $$ Ax + By + Cz = D $$
+
+(We didn't flip the sign of D because it's just some constant that we'll figure out later.)
+
+Here's another way to think of this formula: given the plane perpendicular to the normal vector
+$\mathbf{n} = (A,B,C)$, and the position vector $\mathbf{v} = (x,y,z)$ (that is, the vector from the
+origin to any point on the plane), then
+
+  $$ \mathbf{n} \cdot \mathbf{v} = D $$
+
+Now to find the intersection with some ray $\mathbf{R}(t) = \mathbf{P} + t\mathbf{d}$. Plugging in
+the ray equation, we get
+
+  $$ \mathbf{n} \cdot ( \mathbf{P} + t \mathbf{d} ) = D $$
+
+Solving for $t$:
+
+  $$ \mathbf{n} \cdot \mathbf{P} + \mathbf{n} \cdot t \mathbf{d}  = D $$
+
+  $$ \mathbf{n} \cdot \mathbf{P} + t(\mathbf{n} \cdot \mathbf{d}) = D $$
+
+  $$ t = \frac{D - \mathbf{n} \cdot \mathbf{P}}{\mathbf{n} \cdot \mathbf{d}} $$
+
+This gives us $t$, which we can plug into the ray equation to find the point of intersection.
+
+Ok, we can find the point of intersection between a ray and a plane. This will help us determine
+whether a ray hits a given quadrilateral, using the plane that contains the (planar) quadrilateral.
+In fact, we can use this approach to define any planar primitive, like triangles and disks (more
+later).
+
+We now have three questions we must answer:
+
+1. How do we define the quadrilateral in three dimensional space?
+2. How do we find the plane that contains that quadrilateral?
+3. How do we orient the ray-plane intersection point within the quadrilateral?
+
+
+Defining the Quadrilateral
+---------------------------
+For our purposes, we'll use three geometric entities to define a quadrilateral. Remember that when
+we say "quadrilateral" here, we're really talking about parallelograms, a specific kind of
+quadrilateral where opposite sides are parallel. These three entities are
+
+1. $\mathbf{O}$, the lower-left corner of the quadrilateral.
+2. $\mathbf{u}$, a vector representing the first side of the quadrilateral.
+   $\mathbf{O} + \mathbf{u}$ gives the lower right corner.
+3. $\mathbf{v}$, a vector representing the second side of the quadrilateral.
+   $\mathbf{O} + \mathbf{v}$ gives the upper left corner.
+
+These values are three-dimensional, even though a quad itself is a two-dimensional object.
```

Comment:
Give examples for all of O, u, v (possibly two different sets)

Reply:
(Maybe I should rename these (C, u, v) for corner.)

Reply:
Done.